### PR TITLE
Table component

### PR DESCRIPTION
**Pull request by builat**
* implementation was remastered after all

### CODE REVIEW

File: src/components/icon/assets/arrow_down.svg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<svg width="16" height="16" viewBox="0 0 16 16"  xmlns="http://www.w3.org/2000/svg">
```
Comment (VLegakis) [inline src/components/icon/assets/arrow_down.svg]: Maybe there should also be an arrow up for toggling sorting order. But ok we can put these details later

File: src/components/table/styled.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import styled from "styled-components"
+import { getSizeBy, getColor } from "../../theme/utils"
+import { Icon } from "../icon"
+import { Button } from "../button"
+
+export const StyledTable = styled.table`
+  width: 100%;
```
Comment (VLegakis) [inline src/components/table/styled.tsx]: I think the table doesn't resize its width once the parent container resizes.

File: src/components/table/table.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import React, { useEffect } from "react"
+import { useTable, useSortBy, useRowSelect, ColumnInstance, Row } from "react-table"
+import { UserHeader } from "./components/UserHeader"
+
+import { StyledTable, StyledThead, StyledRow, RowBox } from "./styled"
+
+interface TablePropsT<T, RT = any> {
```
Comment (VLegakis) [inline src/components/table/table.tsx]: This is an Interface (no T required in the name)

File: src/components/table/table.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  columns: RT
+  data: T[]
+  sortedBy?: string[]
+}
+
+function RTable<T extends object>({
+  columns,
+  data,
+  sortedBy = [],
+  selectedItemsClb,
```
Comment (VLegakis) [inline src/components/table/table.tsx]: What does Clb stand for?

Comment (builat) [reply src/components/table/table.tsx]: callbcack

File: src/components/icon/assets/arrow_down.svg (hunk 1 above)
Comment (builat) [reply src/components/icon/assets/arrow_down.svg]: May be, but it is all I have at moment.

image

**Comment by builat** [comment]
I can add example of infinit scrolling, but it probably not the part of the table, but part of `row` component inside

File: src/components/icon/assets/arrow_down.svg (hunk 1 above)
Comment (VLegakis) [reply src/components/icon/assets/arrow_down.svg]: Ok. so if sorting order changes the arrow remains the same right? We can fix that in some next iteration

**Pull request merged**